cmd/query/app/server.go

```diff
@@ -235,17 +239,31 @@ func createHTTPServer(
 
 	apiHandler.RegisterRoutes(r)
 	var handler http.Handler = r
-	handler = responseHeadersHandler(handler, queryOpts.HTTP.ResponseHeaders)
 	if queryOpts.BearerTokenPropagation {
 		handler = bearertoken.PropagationHandler(telset.Logger, handler)
 	}
-	handler = handlers.CompressHandler(handler)
 	recoveryHandler := recoveryhandler.NewRecoveryHandler(telset.Logger, true)
+	handler = recoveryHandler(handler)
+
+	return handler, r
```

Comment:
I don't like this too much, what's the reason we cannot add static handler here? Is it being traced today? I think we tried to avoid tracing requests for ui assets.

Reply:
@yurishkuro i was getting this error when trying to return the closer there https://github.com/jaegertracing/jaeger/pull/6121#discussion_r1817435589